src/result.js

```diff
@@ -33,38 +33,30 @@ var enemyDefenseType = GlobalConst.enemyDefenseType;
 var _ua = GlobalConst._ua;
 var getElementColorLabel = GlobalConst.getElementColorLabel;
 
-var {
-    isCosmos, isDarkOpus, isHollowsky, isValidResult, checkNumberOfRaces,
-    calcCombinations, calcOneCombination,
-    getTesukatoripokaAmount, getTotalBuff,
-    getInitialTotals, getTypeBonus, getTypeBonusStr, calcCriticalDeviation
-} = require('./global_logic.js');
+var {checkNumberOfRaces, getTesukatoripokaAmount, getTypeBonus, getTypeBonusStr, calcCriticalDeviation} = require('./global_logic.js');
 const ResultWorker = require('worker-loader!./calculate_result_worker.js');
 
 var ResultList = CreateClass({
-    calculateResult: function ({ profile, armlist, summon, chara, sortKey }) {
-      const { previousArmlist, previousCombinations } = this.state;
-      this.setState({
-        calculatingResult: true
-      });
-      const worker = new ResultWorker();
-      worker.onmessage = ({ data: { result, previousArmlist, previousCombinations} }) => {
+    calculateResult: function ({profile, armlist, summon, chara, sortKey}) {
+        let worker;
+        const {previousArmlist, previousCombinations} = this.state;
         this.setState({
-          result,
-          previousArmlist: previousArmlist || this.state.previousArmlist,
-          previousCombinations: previousCombinations || this.state.previousCombinations,
-          calculatingResult: false
+            calculatingResult: true
         });
-      }
-      worker.postMessage({
-        profile,
-        armlist,
-        summon,
-        chara,
-        sortKey,
-        previousArmlist,
-        previousCombinations
-      });
+        if (typeof (worker) == "undefined") {
+            worker = new ResultWorker();
+        }
+        worker.onmessage = ({data: {result, previousArmlist, previousCombinations}}) => {
+            this.setState({
+                result,
+                previousArmlist: previousArmlist || this.state.previousArmlist,
+                previousCombinations: previousCombinations || this.state.previousCombinations,
+                calculatingResult: false
+            });
+            worker.terminate();
+            worker = undefined;
```

Comment:
terminate() onmessage will not solve memory leak issue?

Reply:
It will, the memory leak happens because workers have message listeners on them. So the garbage collector can't destroy them and free memory. Terminate() closes the listeners from the outside main thread. The worker can be stopped after the result got returned on that callback.

Reply:
there were two topics,
It's ok about solve the memory leak when task was finished. ( tested :ok: )
How about memory usage on when many worker running
the situation like #404 reported still can be happen.

no memory leak but kind of resource exhaustion.

Can I approve this PR first? then I will report that as another issue.
at least works good on general use.

Reply:
Memory is less of an issue than the computation time. If you have many many combinations, you will run in bottleneck issues regardless of that new solution or the old code. The bottleneck is just not as worse as we currently have with everything running on the main thread and all the computation locking the user.

The best solution would be using a thread pool from like threads.js, but we still have to what is already done here making the calculation into a worker function. But for the current state the worker solution is enough, they should not run that long.